File: samples/client/3_0_3_unit_test/java/src/test/java/org/openapijsonschematools/client/components/schemas/PatternIsNotAnchoredTest.java
package org.openapijsonschematools.client.components.schemas;

import org.junit.Test;
import org.openapijsonschematools.client.configurations.JsonSchemaKeywordFlags;
import org.openapijsonschematools.client.configurations.SchemaConfiguration;

public class PatternIsNotAnchoredTest {
    static final SchemaConfiguration configuration = new SchemaConfiguration(JsonSchemaKeywordFlags.ofNone());

    @Test
    public void testMatchesASubstringPasses() {
        // matches a substring
        final var schema = PatternIsNotAnchored.PatternIsNotAnchored1.getInstance();
        schema.validate(
            "xxaayy",
            configuration
        );
    }
}

File: samples/client/3_0_3_unit_test/kotlin/src/test/kotlin/org/openapijsonschematools/client/components/schemas/PatternIsNotAnchoredTest.kt
package org.openapijsonschematools.client.components.schemas

import org.openapijsonschematools.client.configurations.JsonSchemaKeyword
import org.openapijsonschematools.client.configurations.SchemaConfiguration
import org.openapijsonschematools.client.exceptions.ValidationException

import kotlin.test.Test
import kotlin.test.assertFailsWith

class PatternIsNotAnchoredTest {
    companion object {
        val configuration = SchemaConfiguration(disabledKeywordFlags=setOf(JsonSchemaKeyword.FORMAT))
    }

    @Test
    fun testMatchesASubstringPasses() {
        // matches a substring
        val schema = PatternIsNotAnchored.PatternIsNotAnchored1.getInstance()
        schema.validate(
            "xxaayy",
            configuration
        )
    }
}
